features/users/operations.py

```diff
@@ -1,61 +1,102 @@
+import os
+import secrets
+
+import features.users.exceptions
+from db.connection import get_session
+
+import datetime
+
 from datetime import datetime, timedelta
 from typing import Union, Any, Type
 
 import bcrypt
 from jose import jwt
 from pydantic import ValidationError
 from sqlalchemy import update, delete, insert
+from fastapi.templating import Jinja2Templates
+from httpx import AsyncClient
+from fastapi import HTTPException
 
 import configuration
 import db.connection
 import features.users.exceptions
 from db.connection import get_session
 from .input_models import RegisterUserInputModel
-from .models import User, Role, UserRole
-
-config = configuration.Config()
+from .models import User, Role, UserRole, ConfirmationToken
+from .constants import TokenTypes
 
 
 def hash_password(password: str) -> bytes:
+    """
+    Hash the password
+    :param password:
+    :return:
+    """
+
     salt = bcrypt.gensalt()
     hashed_password = bcrypt.hashpw(password.encode("utf-8"), salt)
     return hashed_password
 
 
 def check_password(user: User, password: str) -> bool:
-    # Check if passwords match (for login)
+    """
+    Check if passwords match (for login)
+
+    :param user:
+    :param password:
+    :return:
+    """
+
     return bcrypt.checkpw(password.encode('utf-8'), user.password)
 
 
 def create_new_user(user: RegisterUserInputModel) -> User:
-    # Create the user in the database
+    """
+    Create user
+
+    :param user:
+    :return:
+    """
+
     with get_session() as session:
-        try:
-            # Check if the username or email already exists
-            if get_user_from_db(username=user.username, email=user.email):
-                # Raise an Exception
-                raise features.users.exceptions.UserAlreadyExists()
-        except features.users.exceptions.UserDoesNotExistException:
-            # Create the new user
-            user.password = hash_password(password=user.password)
-            db_user = User(username=user.username, email=user.email, password=user.password)
-            session.add(db_user)
-            session.commit()
-            session.refresh(db_user)
+        if get_user_from_db(username=user.username, email=user.email):
```

Comment:
You have to return the try block because It will raise a error when we are creating the new user (it won't be in the db and get_user_from_db will raise user not found)